### PR TITLE
Uptade pytest-timeout and x265 derivations

### PR DESCRIPTION
**Pull request by jpgu-epam**
###### Motivation for this change

Update the following derivations and remove old patches on those:

- pytest-timeout: 1.3.3 -> 1.4.2
- x265: 3.2 -> 3.4

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/x265/default.nix [hunk 1]
```diff
@@ -23,27 +23,18 @@ let
...
-  src = fetchurl {
-    urls = [
-      "https://get.videolan.org/x265/x265_${version}.tar.gz"
-      "ftp://ftp.videolan.org/pub/videolan/x265/x265_${version}.tar.gz"
-    ];
-    sha256 = "0fqkhfhr22gzavxn60cpnj3agwdf5afivszxf3haj5k1sny7jk9n";
+  src = fetchFromGitHub {
+    owner  = "videolan";
+    repo   = "x265";
+    rev    = "${version}";
```
Comment (jonringer) [inline pkgs/development/libraries/x265/default.nix:31]:
```suggestion
    rev    = version;
```

**Review by jonringer — COMMENTED** [review]
I notice that your branch name mentions nixos-20.03. Unless it's a security update, this likely won't get backported (due to potential breakages)

**Comment by jpgu-epam** [comment]
> I notice that your branch name mentions nixos-20.03. Unless it's a security update, this likely won't get backported (due to potential breakages)

So, should I create a PR to nixpkgs-20.03 as well?

**Comment by SuperSandro2000** [comment]
> So, should I create a PR to nixpkgs-20.03 as well?

20.03 is EOL so this is to late.

**Comment by SuperSandro2000** [comment]
Please fix the merge conflict.

**Comment by tomberek** [comment]
both bumps have reached master. src seems to be correct in master as well.

**Pull request closed without merging**